src/js/select2/selection/search.js

```diff
@@ -100,6 +100,7 @@ define([
   };
 
   Search.prototype.update = function (decorated, data) {
+    var had_focus = this.$search.is(":focus");
```

Comment:
Also, I wouldn't mind comparing `this.$search[0] == document.activeElement` to avoid the Sizzle requirement.